beluga/include/beluga/sensor/data/laser_scan.hpp

```diff
@@ -0,0 +1,70 @@
+// Copyright 2023 Ekumen, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BELUGA_SENSOR_DATA_LASER_SCAN_HPP
+#define BELUGA_SENSOR_DATA_LASER_SCAN_HPP
+
+#include <cmath>
+
+#include <ciabatta/ciabatta.hpp>
+
+#include <sophus/types.hpp>
+
+#include <range/v3/view/filter.hpp>
+#include <range/v3/view/transform.hpp>
+#include <range/v3/view/zip.hpp>
+
+/**
+ * \file
+ * \brief Implementation of a laser scan interface.
+ */
+
+namespace beluga {
+
+/// Base laser scan implementation.
+/**
+ * TODO(nahuel): Complete documentation.
+ */
+template <typename Derived>
+class BaseLaserScan : public ciabatta::ciabatta_top<Derived> {
+ public:
+  /// View points in cartesian coordinates.
+  [[nodiscard]] auto points_in_cartesian_coordinates() const {
```

Comment:
@nahueespinosa in what frame? I see we define an `origin` but nothing makes use of the transform.

Reply:
Added documentation in https://github.com/Ekumen-OS/beluga/pull/271/commits/15d93bcdc91b64f931976ee5f4d232f79c963564.